### ISSUE TITLE
GitLab Collector is not running

### ISSUE THREAD

**Issue by hackeys** [capitalone/Hygieia#957]
Hi , 

Whenever i run gitlab collector , i get following error .
```
2016-11-17 16:15:38,434 INFO  org.mongodb.driver.cluster - Monitor thread successfully connected to server with description ServerDescription{address=10.14.34.73:27017, type=STANDALONE, state=CONNECTED, ok=true, version=ServerVersion{versionList=[3, 2, 10]}, minWireVersion=0, maxWireVersion=4, electionId=null, maxDocumentSize=16777216, roundTripTimeNanos=912360}
2016-11-17 16:15:39,316 INFO  o.s.j.e.a.AnnotationMBeanExporter - Registering beans for JMX exposure on startup
2016-11-17 16:15:39,327 INFO  com.capitalone.dashboard.Application - Started Application in 9.02 seconds (JVM running for 11.151)
2016-11-17 16:18:00,000 INFO  c.c.d.collector.CollectorTask - Running Collector: Gitlab
2016-11-17 16:18:00,009 INFO  c.c.d.collector.CollectorTask - -----------------------------------
2016-11-17 16:18:00,010 INFO  c.c.d.collector.CollectorTask - Starting...
2016-11-17 16:18:00,010 INFO  c.c.d.collector.CollectorTask - -----------------------------------
2016-11-17 16:18:00,118 ERROR o.s.s.s.TaskUtils$LoggingErrorHandler - Unexpected error occurred in scheduled task.
java.lang.NullPointerException: null
        at com.capitalone.dashboard.collecteur.DefaultGitlabGitClient.normalizeUrl(DefaultGitlabGitClient.java:239) ~[gitlab-git-collector-2.0.4-SNAPSHOT.jar!/:2.0.4-SNAPSHOT]
        at com.capitalone.dashboard.collecteur.DefaultGitlabGitClient.getCommits(DefaultGitlabGitClient.java:68) ~[gitlab-git-collector-2.0.4-SNAPSHOT.jar!/:2.0.4-SNAPSHOT]
        at com.capitalone.dashboard.collecteur.GitlabGitCollectorTask.collect(GitlabGitCollectorTask.java:76) ~[gitlab-git-collector-2.0.4-SNAPSHOT.jar!/:2.0.4-SNAPSHOT]
        at com.capitalone.dashboard.collector.CollectorTask.run(CollectorTask.java:63) ~[core-2.0.4-SNAPSHOT.jar!/:2.0.4-SNAPSHOT]
        at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54) ~[spring-context-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
        at org.springframework.scheduling.concurrent.ReschedulingRunnable.run(ReschedulingRunnable.java:81) [spring-context-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [na:1.8.0_45]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0_45]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [na:1.8.0_45]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [na:1.8.0_45]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_45]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_45]
        at java.lang.Thread.run(Thread.java:745) [na:1.8.0_45]
```

Below is my property file 
```properties
dbname=db
dbusername=db
dbpassword=dbpass
dbhost=10.14.34.73
dbport=27017
dbreplicaset=false
dbhostport=10.14.34.73:27017
gitlab.cron=0 0/3 * * * *
gitlab.branche=master
gitlab.apiToken=BrXVX-EFBtkq3Jkv6
gitlab.saveLog=true
gitlab.server=http://10.14.34.73:8181
```

**Comment by rob-miller-777** [comment]
Do you have the latest code pulled down?  I made some changes last week that might help with your issue.  Based on your stack trace, it looks like you are on the old version. I would try pulling the latest and updating your properties file to the new values to see if that works.  Let me know if you have questions.

**Comment by mrpudn** [comment]
See also #910

**Comment by hackeys** [comment]
Hi All , 

Thanks . Per suggestion , i cloned new master code and now getting new error .

Can anybody please suggest me on how do i mention port in .properties file .

**My Current Properties File looks like :**
 cat application.properties
dbname=db
dbusername=db
dbpassword=dbpass
dbhost=10.14.34.73
dbport=27017
dbreplicaset=false
dbhostport=10.14.34.73:27017
gitlab.cron=0 0/3 * * * *
gitlab.branche=master
gitlab.apiToken=BrXVX-EFBtkq3Jkv6JCD
gitlab.saveLog=true
gitlab.server=http://10.14.34.73:8181
gitlab.host=10.14.34.73
gitlab.port=8181
gitlab.selfSignedCertificate=false
gitlab.commitThresholdDays=15

6-11-24 00:30:00,004 INFO  c.c.d.collector.CollectorTask - -----------------------------------
2016-11-24 00:30:00,188 ERROR o.s.s.s.TaskUtils$LoggingErrorHandler - Unexpected error occurred in scheduled task.
org.springframework.web.client.ResourceAccessException: I/O error on GET request for "https://10.14.34.73/api/v3/projects/root%2FHygieia/repository/commits/?ref_name&since=2016-11-24T00:20:00Z&per_page=100":Connect to 10.14.34.73:443 [/10.14.34.73] failed: Connection refused; nested exception is org.apache.http.conn.HttpHostConnectException: Connect to 10.14.34.73:443 [/10.14.34.73] failed: Connection refused
        at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:580) ~[spring-web-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
        at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:545) ~[spring-web-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
        at org.springframework.web.client.RestTemplate.exchange(RestTemplate.java:466) ~[spring-web-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
        at com.capitalone.dashboard.collecteur.DefaultGitlabGitClient.makeRestCall(DefaultGitlabGitClient.java:92) ~[gitlab-git-collector-2.0.4-SNAPSHOT.jar!/:2.0.4-SNAPSHOT]
        at com.capitalone.dashboard.collecteur.DefaultGitlabGitClient.getCommits(DefaultGitlabGitClient.java:69) ~[gitlab-git-collector-2.0.4-SNAPSHOT.jar!/:2.0.4-SNAPSHOT]
        at com.capitalone.dashboard.collecteur.GitlabGitCollectorTask.collect(GitlabGitCollectorTask.java:98) ~[gitlab-git-collector-2.0.4-SNAPSHOT.jar!/:2.0.4-SNAPSHOT]
        at com.capitalone.dashboard.collector.CollectorTask.run(CollectorTask.java:63) ~[core-2.0.4-SNAPSHOT.jar!/:2.0.4-SNAPSHOT]
        at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54) ~[spring-context-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
        at org.springframework.scheduling.concurrent.ReschedulingRunnable.run(ReschedulingRunnable.java:81) [spring-context-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [na:1.8.0_101]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0_101]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [na:1.8.0_101]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [na:1.8.0_101]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_101]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_101]
        at java.lang.Thread.run(Thread.java:745) [na:1.8.0_101]
Caused by: org.apache.http.conn.HttpHostConnectException: Connect to 10.14.34.73:443 [/10.14.34.73] failed: Connection refused
        at org.apache.http.impl.conn.HttpClientConnectionOperator.connect(HttpClientConnectionOperator.java:142) ~[httpclient-4.3.6.jar!/:4.3.6]
        at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:319) ~[httpclient-4.3.6.jar!/:4.3.6]
        at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:363) ~[httpclient-4.3.6.jar!/:4.3.6]
        at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:219) ~[httpclient-4.3.6.jar!/:4.3.6]
        at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:195) ~[httpclient-4.3.6.jar!/:4.3.6

**Comment by rob-miller-777** [comment]
As far as the properties, the only ones that are used are below.  If you are using port 8181 for your gitlab instance, try adding that to the end of the host as seen below. If your instance of gitlab has a self signed certificate, you should should try setting that property to true as well. 
 
gitlab.cron=0 0/3 * * * *
gitlab.branche=master
gitlab.apiToken=BrXVX-EFBtkq3Jkv6JCD
gitlab.host=10.14.34.73:8181
gitlab.selfSignedCertificate=false
gitlab.commitThresholdDays=15

**Comment by rob-miller-777** [comment]
And "branche" is also no longer a property. You set that from the UI now.

**Comment by hackeys** [comment]
I get error when i append port number to the host like below .
gitlab.host=10.14.34.73:8181

**Getting below error :**


14,991 INFO  o.s.j.e.a.AnnotationMBeanExporter - Registering beans for JMX exposure on startup
2016-11-24 01:22:14,999 INFO  com.capitalone.dashboard.Application - Started Application in 3.74 seconds (JVM running for 4.369)
2016-11-24 01:24:00,000 INFO  c.c.d.collector.CollectorTask - Running Collector: Gitlab
2016-11-24 01:24:00,003 INFO  c.c.d.collector.CollectorTask - -----------------------------------
2016-11-24 01:24:00,003 INFO  c.c.d.collector.CollectorTask - Starting...
2016-11-24 01:24:00,003 INFO  c.c.d.collector.CollectorTask - -----------------------------------
2016-11-24 01:24:00,050 ERROR o.s.s.s.TaskUtils$LoggingErrorHandler - Unexpected error occurred in scheduled task.
java.lang.IllegalArgumentException: Invalid character ':' for HOST_IPV4 in "10.14.34.73:8181"
        at org.springframework.web.util.HierarchicalUriComponents.verifyUriComponent(HierarchicalUriComponents.java:304) ~[spring-web-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
        at org.springframework.web.util.HierarchicalUriComponents.verify(HierarchicalUriComponents.java:269) ~[spring-web-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
        at org.springframework.web.util.HierarchicalUriComponents.<init>(HierarchicalUriComponents.java:88) ~[spring-web-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
        at org.springframework.web.util.UriComponentsBuilder.build(UriComponentsBuilder.java:364) ~[spring-web-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
        at com.capitalone.dashboard.collecteur.GitlabUrlUtility.buildApiUrl(GitlabUrlUtility.java:61) ~[gitlab-git-collector-2.0.4-SNAPSHOT.jar!/:2.0.4-SNAPSHOT]
        at com.capitalone.dashboard.collecteur.DefaultGitlabGitClient.getCommits(DefaultGitlabGitClient.java:61) ~[gitlab-git-collector-2.0.4-SNAPSHOT.jar!/:2.0.4-SNAPSHOT]
        at com.capitalone.dashboard.collecteur.GitlabGitCollectorTask.collect(GitlabGitCollectorTask.java:98) ~[gitlab-git-collector-2.0.4-SNAPSHOT.jar!/:2.0.4-SNAPSHOT]
        at com.capitalone.dashboard.collector.CollectorTask.run(CollectorTask.java:63) ~[core-2.0.4-SNAPSHOT.jar!/:2.0.4-SNAPSHOT]
        at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54) ~[spring-context-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
        at org.springframework.scheduling.concurrent.ReschedulingRunnable.run(ReschedulingRunnable.java:81) [spring-context-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [na:1.8.0_101]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0_101]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [na:1.8.0_101]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [na:1.8.0_101]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_101]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_101]
        at java.lang.Thread.run(Thread.java:745) [na:1.8.0_101]

**Comment by rob-miller-777** [comment]
Yeah, had a feeling that might happen. What happens if you try to hit the url it's building in a browser.  https://10.14.34.73/api/v3/projects/root%2FHygieia/repository/commits/?ref_name&since=2016-11-24T00:20:00Z&per_page=100"

**Comment by hackeys** [comment]
![error](https://cloud.githubusercontent.com/assets/2296676/20576889/cf596e20-b1e5-11e6-92c8-89659d750031.png)
I get [] as result when browsed 

http://10.14.34.73:8181/api/v3/projects/root%2FHygieia/repository/commits/?ref_name&since=2016-11-24T00:20:00Z&per_page=100

Over browser .
Attaching screenshot.

However , 

I dont find anything on browser when run 

https://10.14.34.73/api/v3/projects/root%2FHygieia/repository/commits/?ref_name&since=2016-11-24T00:20:00Z&per_page=100"


My GitLab is hosted on http and not https . It is hosted on port 8181 .

**Comment by rob-miller-777** [comment]
Got ya, think a couple small changes have to be made to the code to make this work.  Currently it will always try to use https, and doesn't support ports.  I hadn't tested it with an IP address and port, but it shouldn't be hard to fix.  I can probably get a pull request out there to fix this issue later next week, probably won't have time this week with Thanksgiving and all.  Sorry about the inconvenience.

**Comment by rob-miller-777** [comment]
My idea would be to add two additional (optional) fields to the properties file, one for protocol, and one for port.  Would probably default the protocol to "https", and omit the port if not given.  All of the logic would go into the class below.  https://github.com/capitalone/Hygieia/blob/master/collectors/scm/gitlab/src/main/java/com/capitalone/dashboard/collecteur/GitlabUrlUtility.java

**Comment by hackeys** [comment]
Sure @rob-miller-777  .. Thanks a lot for quick response .

Appreciate your effort and time . Have a fun filled thanksgiving .

**Comment by rob-miller-777** [comment]
@hackeys I made the changes that should fix your issue.  Just waiting on some approvals from our end before I can submit the pull request.  Hopefully have that open in the next day or two.  I'll put a link in here when I open the pull request.

**Comment by vetrivelan25** [comment]
Hi ,
We need to download latest package again or just add the lines as you mentioned in the above 3 files..
added support for custon protocol and port
updated readme
small readme fix

**Comment by vanjos** [comment]
Hi all,
I have the up-to-date code (from 4 days ago) and have been able to build the UI and API, but cannot get the Gitlab SCM collector to work properly.  My major error begins with

`Error creating bean with name 'gitlabSettings': Could not bind properties to [unknown]`

I'm running this off of docker images, I did not want to do this locally:

> application.properties
`===========================================
Properties file created Tue Apr 11 16:53:05 UTC 2017:  hygieia-gitlab-scm-collector.properties
Note: passwords/keys hidden
===========================================
#Database Name
dbname=dashboard
#Database HostName - default is localhost
dbhost=172.17.0.2
#Database Port - default is 27017
dbport=27017
#Database Username - default is blank
dbusername=db
#Database Password - default is blank
dbpassword=****
#Collector schedule (required)
gitlab.cron=0 0/5 * * * *
#Gitlab host (optional, defaults to "gitlab.com")
gitlab.host=gitlab.com
#Gitlab protocol (optional, defaults to "http")
gitlab.protocol=https
#Gitlab port (optional, defaults to protocol default port)
gitlab.port=
#Gitlab path (optional, defaults to no path)
gitlab.path=deeplearni.ng
#Gitlab API Token (required, access token can be retrieved through gitlab, collector will have the permissions of the user associated to the token)
gitlab.apiToken=****
#Maximum number of days to go back in time when fetching commits
gitlab.commitThresholdDays=15
#Gitlab Instance using self signed certificate
gitlab.selfSignedCertificate=false`

and when I check the docker logs:
> docker logs hygieia-gitlab -f
`16:53:07,385 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback.groovy]
16:53:07,390 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback-test.xml]
16:53:07,390 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Found resource [logback.xml] at [jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/logback.xml]
16:53:07,407 |-WARN in ch.qos.logback.classic.LoggerContext[default] - Resource [logback.xml] occurs multiple times on the classpath.
16:53:07,407 |-WARN in ch.qos.logback.classic.LoggerContext[default] - Resource [logback.xml] occurs at [jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/core-2.0.5-SNAPSHOT.jar!/logback.xml]
16:53:07,407 |-WARN in ch.qos.logback.classic.LoggerContext[default] - Resource [logback.xml] occurs at [jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/logback.xml]
16:53:07,780 |-INFO in ch.qos.logback.core.joran.spi.ConfigurationWatchList@231f2b44 - URL [jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/logback.xml] is not of type file
16:53:08,414 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - debug attribute not set
16:53:08,470 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
16:53:08,575 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [STDOUT]
16:53:08,980 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
16:53:10,024 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.rolling.RollingFileAppender]
16:53:10,034 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [ROLLING]
16:53:10,580 |-INFO in c.q.l.core.rolling.TimeBasedRollingPolicy - No compression will be used
16:53:10,582 |-INFO in c.q.l.core.rolling.TimeBasedRollingPolicy - Will use the pattern logs/gitlab-git-collector-%d{yyyy-MM-dd}.%i.log for the active file
16:53:10,600 |-INFO in ch.qos.logback.core.rolling.SizeAndTimeBasedFNATP@4d35f51b - The date pattern is 'yyyy-MM-dd' from file name pattern 'logs/gitlab-git-collector-%d{yyyy-MM-dd}.%i.log'.
16:53:10,600 |-INFO in ch.qos.logback.core.rolling.SizeAndTimeBasedFNATP@4d35f51b - Roll-over at midnight.
16:53:10,669 |-INFO in ch.qos.logback.core.rolling.SizeAndTimeBasedFNATP@4d35f51b - Setting initial period to Tue Apr 11 16:53:10 UTC 2017
16:53:10,691 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
16:53:10,692 |-INFO in ch.qos.logback.core.rolling.RollingFileAppender[ROLLING] - Active log file name: logs/gitlab-git-collector-2017-04-11.0.log
16:53:10,692 |-INFO in ch.qos.logback.core.rolling.RollingFileAppender[ROLLING] - File property is set to [null]
16:53:10,693 |-INFO in ch.qos.logback.classic.joran.action.RootLoggerAction - Setting level of ROOT logger to INFO
16:53:10,693 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [ROLLING] to Logger[ROOT]
16:53:10,699 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [STDOUT] to Logger[ROOT]
16:53:10,699 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - End of configuration.
16:53:10,700 |-INFO in ch.qos.logback.classic.joran.JoranConfigurator@53fcf97b - Registering current configuration as safe fallback point

  _    ___     _______ _____ ______ _____                _____ _____ _______ _               ____         _____ ____  _      _      ______ _____ _______ ____  _____  
 | |  | \ \   / / ____|_   _|  ____|_   _|   /\         / ____|_   _|__   __| |       /\    |  _ \       / ____/ __ \| |    | |    |  ____/ ____|__   __/ __ \|  __ \ 
 | |__| |\ \_/ / |  __  | | | |__    | |    /  \ ______| |  __  | |    | |  | |      /  \   | (_) |_____| |   | |  | | |    | |    | |__ | |       | | | |  | | |__) |
 |  __  | \   /| | |_ | | | |  __|   | |   / /\ \______| | |_ | | |    | |  | |     / /\ \  |  _ < _____| |   | |  | | |    | |    |  __|| |       | | | |  | |  _  / 
 | |  | |  | | | |__| |_| |_| |____ _| |_ / ____ \     | |__| |_| |_   | |  | |__  / ____ \ | (_) |     | |___| |__| | |____| |____| |___| |____   | | | |__| | | \ \ 
 |_|  |_|  |_|  \_____|_____|______|_____/_/    \_\     \_____|_____|  |_|  |____|/_/    \_\|____/       \_____\____/|______|______|______\_____|  |_|  \____/|_|  \_\



2017-04-11 16:53:13,032 INFO  com.capitalone.dashboard.Application - Starting Application v2.0.5-SNAPSHOT on 16f6968917c1 with PID 24 (/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar started by root in /hygieia)
2017-04-11 16:53:13,352 INFO  o.s.c.a.AnnotationConfigApplicationContext - Refreshing org.springframework.context.annotation.AnnotationConfigApplicationContext@1457ac73: startup date [Tue Apr 11 16:53:13 UTC 2017]; root of context hierarchy
2017-04-11 16:53:23,185 INFO  c.u.j.EnableEncryptablePropertySourcesPostProcessor - Post-processing PropertySource instances
2017-04-11 16:53:23,407 INFO  c.u.j.c.StringEncryptorConfiguration - String Encryptor custom Bean not found with name 'jasyptStringEncryptor'. Initializing String Encryptor based on properties with name 'jasyptStringEncryptor'
2017-04-11 16:53:23,462 INFO  c.u.j.EnableEncryptablePropertySourcesPostProcessor - Converting PropertySource commandLineArgs [org.springframework.core.env.SimpleCommandLinePropertySource] to EncryptableEnumerablePropertySourceWrapper
2017-04-11 16:53:23,465 INFO  c.u.j.EnableEncryptablePropertySourcesPostProcessor - Converting PropertySource systemProperties [org.springframework.core.env.MapPropertySource] to EncryptableMapPropertySourceWrapper
2017-04-11 16:53:23,475 INFO  c.u.j.EnableEncryptablePropertySourcesPostProcessor - Converting PropertySource systemEnvironment [org.springframework.core.env.SystemEnvironmentPropertySource] to EncryptableMapPropertySourceWrapper
2017-04-11 16:53:23,475 INFO  c.u.j.EnableEncryptablePropertySourcesPostProcessor - Converting PropertySource random [org.springframework.boot.context.config.RandomValuePropertySource] to EncryptablePropertySourceWrapper
2017-04-11 16:53:23,476 INFO  c.u.j.EnableEncryptablePropertySourcesPostProcessor - Converting PropertySource applicationConfig: [file:/hygieia/hygieia-gitlab-scm-collector.properties] [org.springframework.core.env.PropertiesPropertySource] to EncryptableMapPropertySourceWrapper
2017-04-11 16:53:23,587 INFO  o.s.b.f.a.AutowiredAnnotationBeanPostProcessor - JSR-330 'javax.inject.Inject' annotation found and supported for autowiring
2017-04-11 16:53:23,883 INFO  o.h.validator.internal.util.Version - HV000001: Hibernate Validator 5.1.3.Final
2017-04-11 16:53:24,429 WARN  o.s.c.a.AnnotationConfigApplicationContext - Exception encountered during context initialization - cancelling refresh attempt
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'gitlabSettings': Could not bind properties to [unknown] (target=gitlab, ignoreInvalidFields=false, ignoreUnknownFields=true, ignoreNestedProperties=false); nested exception is java.lang.IllegalStateException: Required Encryption configuration property missing: jasypt.encryptor.password
	at org.springframework.boot.context.properties.ConfigurationPropertiesBindingPostProcessor.postProcessBeforeInitialization(ConfigurationPropertiesBindingPostProcessor.java:303) ~[spring-boot-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	at org.springframework.boot.context.properties.ConfigurationPropertiesBindingPostProcessor.postProcessBeforeInitialization(ConfigurationPropertiesBindingPostProcessor.java:250) ~[spring-boot-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsBeforeInitialization(AbstractAutowireCapableBeanFactory.java:408) ~[spring-beans-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1566) ~[spring-beans-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:539) ~[spring-beans-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:476) ~[spring-beans-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:303) ~[spring-beans-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230) ~[spring-beans-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:299) ~[spring-beans-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:194) ~[spring-beans-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:755) ~[spring-beans-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:757) ~[spring-context-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:480) ~[spring-context-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:686) [spring-boot-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:320) [spring-boot-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:957) [spring-boot-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:946) [spring-boot-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	at com.capitalone.dashboard.Application.main(Application.java:16) [gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/:2.0.5-SNAPSHOT]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_111]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_111]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_111]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_111]
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:53) [gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/:2.0.5-SNAPSHOT]
	at java.lang.Thread.run(Thread.java:745) [na:1.8.0_111]
Caused by: java.lang.IllegalStateException: Required Encryption configuration property missing: jasypt.encryptor.password
	at com.ulisesbocchio.jasyptspringboot.configuration.StringEncryptorConfiguration.getRequiredProperty(StringEncryptorConfiguration.java:80) ~[jasypt-spring-boot-1.9.jar!/:na]
	at com.ulisesbocchio.jasyptspringboot.configuration.StringEncryptorConfiguration.lambda$static$0(StringEncryptorConfiguration.java:41) ~[jasypt-spring-boot-1.9.jar!/:na]
	at com.ulisesbocchio.jasyptspringboot.encryptor.LazyStringEncryptor$SingletonFunction.lambda$new$1(LazyStringEncryptor.java:50) ~[jasypt-spring-boot-1.9.jar!/:na]
	at com.ulisesbocchio.jasyptspringboot.encryptor.LazyStringEncryptor$SingletonFunction.apply(LazyStringEncryptor.java:63) ~[jasypt-spring-boot-1.9.jar!/:na]
	at com.ulisesbocchio.jasyptspringboot.encryptor.LazyStringEncryptor.decrypt(LazyStringEncryptor.java:32) ~[jasypt-spring-boot-1.9.jar!/:na]
	at org.jasypt.properties.PropertyValueEncryptionUtils.decrypt(PropertyValueEncryptionUtils.java:72) ~[jasypt-1.9.2.jar!/:na]
	at com.ulisesbocchio.jasyptspringboot.EncryptablePropertySource.getProperty(EncryptablePropertySource.java:19) ~[jasypt-spring-boot-1.9.jar!/:na]
	at com.ulisesbocchio.jasyptspringboot.wrapper.EncryptableMapPropertySourceWrapper.getProperty(EncryptableMapPropertySourceWrapper.java:28) ~[jasypt-spring-boot-1.9.jar!/:na]
	at org.springframework.boot.bind.PropertySourcesPropertyValues.getEnumerableProperty(PropertySourcesPropertyValues.java:140) ~[spring-boot-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	at org.springframework.boot.bind.PropertySourcesPropertyValues.processEnumerablePropertySource(PropertySourcesPropertyValues.java:124) ~[spring-boot-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	at org.springframework.boot.bind.PropertySourcesPropertyValues.processPropertySource(PropertySourcesPropertyValues.java:106) ~[spring-boot-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	at org.springframework.boot.bind.PropertySourcesPropertyValues.<init>(PropertySourcesPropertyValues.java:94) ~[spring-boot-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	at org.springframework.boot.bind.PropertiesConfigurationFactory.getPropertyValues(PropertiesConfigurationFactory.java:281) ~[spring-boot-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	at org.springframework.boot.bind.PropertiesConfigurationFactory.doBindPropertiesToTarget(PropertiesConfigurationFactory.java:252) ~[spring-boot-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	at org.springframework.boot.bind.PropertiesConfigurationFactory.bindPropertiesToTarget(PropertiesConfigurationFactory.java:227) ~[spring-boot-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	at org.springframework.boot.context.properties.ConfigurationPropertiesBindingPostProcessor.postProcessBeforeInitialization(ConfigurationPropertiesBindingPostProcessor.java:296) ~[spring-boot-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	... 23 common frames omitted
2017-04-11 16:53:24,495 INFO  o.s.b.l.ClasspathLoggingApplicationListener - Application failed to start with classpath: [jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/core-2.0.5-SNAPSHOT.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/spring-boot-1.2.5.RELEASE.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/spring-test-4.1.7.RELEASE.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/spring-data-mongodb-1.8.0.RELEASE.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/spring-expression-4.1.7.RELEASE.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/spring-data-commons-1.11.0.RELEASE.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/slf4j-api-1.7.12.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/jcl-over-slf4j-1.7.12.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/querydsl-jpa-3.6.0.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/querydsl-core-3.6.0.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/guava-18.0.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/jsr305-1.3.9.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/mysema-commons-lang-0.2.4.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/bridge-method-annotation-1.13.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/querydsl-apt-3.6.0.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/querydsl-codegen-3.6.0.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/codegen-0.6.5.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/ecj-4.3.1.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/javax.inject-1.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/querydsl-mongodb-3.6.0.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/jasypt-spring-boot-starter-1.9.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/jasypt-spring-boot-1.9.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/jasypt-1.9.2.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/validation-api-1.1.0.Final.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/hibernate-validator-5.1.3.Final.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/jboss-logging-3.1.3.GA.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/classmate-1.0.0.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/javax.el-api-2.2.4.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/commons-beanutils-1.9.2.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/commons-collections-3.2.1.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/spring-boot-starter-data-mongodb-1.2.5.RELEASE.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/spring-boot-starter-1.2.5.RELEASE.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/spring-boot-autoconfigure-1.2.5.RELEASE.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/spring-boot-starter-logging-1.2.5.RELEASE.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/jul-to-slf4j-1.7.12.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/log4j-over-slf4j-1.7.12.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/logback-classic-1.1.3.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/logback-core-1.1.3.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/snakeyaml-1.14.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/mongo-java-driver-3.0.2.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/spring-core-4.1.7.RELEASE.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/spring-tx-4.1.7.RELEASE.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/junit-4.12.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/hamcrest-core-1.3.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/spring-web-4.1.7.RELEASE.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/spring-aop-4.1.7.RELEASE.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/aopalliance-1.0.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/spring-beans-4.1.7.RELEASE.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/spring-context-4.1.7.RELEASE.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/json-simple-1.1.1.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/commons-lang3-3.3.2.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/commons-codec-1.9.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/httpclient-4.3.6.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/httpcore-4.3.3.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/joda-time-2.7.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/joda-convert-1.8.1.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/jackson-databind-2.5.0.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/jackson-annotations-2.5.0.jar!/, jar:file:/hygieia/gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/lib/jackson-core-2.5.0.jar!/]
2017-04-11 16:53:24,576 INFO  o.s.b.a.l.AutoConfigurationReportLoggingInitializer - 

Error starting ApplicationContext. To display the auto-configuration report enable debug logging (start with --debug)


2017-04-11 16:53:24,577 ERROR o.s.boot.SpringApplication - Application startup failed
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'gitlabSettings': Could not bind properties to [unknown] (target=gitlab, ignoreInvalidFields=false, ignoreUnknownFields=true, ignoreNestedProperties=false); nested exception is java.lang.IllegalStateException: Required Encryption configuration property missing: jasypt.encryptor.password
	at org.springframework.boot.context.properties.ConfigurationPropertiesBindingPostProcessor.postProcessBeforeInitialization(ConfigurationPropertiesBindingPostProcessor.java:303) ~[spring-boot-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	at org.springframework.boot.context.properties.ConfigurationPropertiesBindingPostProcessor.postProcessBeforeInitialization(ConfigurationPropertiesBindingPostProcessor.java:250) ~[spring-boot-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsBeforeInitialization(AbstractAutowireCapableBeanFactory.java:408) ~[spring-beans-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1566) ~[spring-beans-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:539) ~[spring-beans-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:476) ~[spring-beans-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:303) ~[spring-beans-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230) ~[spring-beans-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:299) ~[spring-beans-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:194) ~[spring-beans-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:755) ~[spring-beans-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:757) ~[spring-context-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:480) ~[spring-context-4.1.7.RELEASE.jar!/:4.1.7.RELEASE]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:686) [spring-boot-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:320) [spring-boot-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:957) [spring-boot-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:946) [spring-boot-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	at com.capitalone.dashboard.Application.main(Application.java:16) [gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/:2.0.5-SNAPSHOT]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_111]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_111]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_111]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_111]
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:53) [gitlab-scm-collector-2.0.5-SNAPSHOT.jar!/:2.0.5-SNAPSHOT]
	at java.lang.Thread.run(Thread.java:745) [na:1.8.0_111]
Caused by: java.lang.IllegalStateException: Required Encryption configuration property missing: jasypt.encryptor.password
	at com.ulisesbocchio.jasyptspringboot.configuration.StringEncryptorConfiguration.getRequiredProperty(StringEncryptorConfiguration.java:80) ~[jasypt-spring-boot-1.9.jar!/:na]
	at com.ulisesbocchio.jasyptspringboot.configuration.StringEncryptorConfiguration.lambda$static$0(StringEncryptorConfiguration.java:41) ~[jasypt-spring-boot-1.9.jar!/:na]
	at com.ulisesbocchio.jasyptspringboot.encryptor.LazyStringEncryptor$SingletonFunction.lambda$new$1(LazyStringEncryptor.java:50) ~[jasypt-spring-boot-1.9.jar!/:na]
	at com.ulisesbocchio.jasyptspringboot.encryptor.LazyStringEncryptor$SingletonFunction.apply(LazyStringEncryptor.java:63) ~[jasypt-spring-boot-1.9.jar!/:na]
	at com.ulisesbocchio.jasyptspringboot.encryptor.LazyStringEncryptor.decrypt(LazyStringEncryptor.java:32) ~[jasypt-spring-boot-1.9.jar!/:na]
	at org.jasypt.properties.PropertyValueEncryptionUtils.decrypt(PropertyValueEncryptionUtils.java:72) ~[jasypt-1.9.2.jar!/:na]
	at com.ulisesbocchio.jasyptspringboot.EncryptablePropertySource.getProperty(EncryptablePropertySource.java:19) ~[jasypt-spring-boot-1.9.jar!/:na]
	at com.ulisesbocchio.jasyptspringboot.wrapper.EncryptableMapPropertySourceWrapper.getProperty(EncryptableMapPropertySourceWrapper.java:28) ~[jasypt-spring-boot-1.9.jar!/:na]
	at org.springframework.boot.bind.PropertySourcesPropertyValues.getEnumerableProperty(PropertySourcesPropertyValues.java:140) ~[spring-boot-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	at org.springframework.boot.bind.PropertySourcesPropertyValues.processEnumerablePropertySource(PropertySourcesPropertyValues.java:124) ~[spring-boot-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	at org.springframework.boot.bind.PropertySourcesPropertyValues.processPropertySource(PropertySourcesPropertyValues.java:106) ~[spring-boot-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	at org.springframework.boot.bind.PropertySourcesPropertyValues.<init>(PropertySourcesPropertyValues.java:94) ~[spring-boot-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	at org.springframework.boot.bind.PropertiesConfigurationFactory.getPropertyValues(PropertiesConfigurationFactory.java:281) ~[spring-boot-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	at org.springframework.boot.bind.PropertiesConfigurationFactory.doBindPropertiesToTarget(PropertiesConfigurationFactory.java:252) ~[spring-boot-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	at org.springframework.boot.bind.PropertiesConfigurationFactory.bindPropertiesToTarget(PropertiesConfigurationFactory.java:227) ~[spring-boot-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	at org.springframework.boot.context.properties.ConfigurationPropertiesBindingPostProcessor.postProcessBeforeInitialization(ConfigurationPropertiesBindingPostProcessor.java:296) ~[spring-boot-1.2.5.RELEASE.jar!/:1.2.5.RELEASE]
	... 23 common frames omitted
java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:53)
	at java.lang.Thread.run(Thread.java:745)
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'gitlabSettings': Could not bind properties to [unknown] (target=gitlab, ignoreInvalidFields=false, ignoreUnknownFields=true, ignoreNestedProperties=false); nested exception is java.lang.IllegalStateException: Required Encryption configuration property missing: jasypt.encryptor.password
	at org.springframework.boot.context.properties.ConfigurationPropertiesBindingPostProcessor.postProcessBeforeInitialization(ConfigurationPropertiesBindingPostProcessor.java:303)
	at org.springframework.boot.context.properties.ConfigurationPropertiesBindingPostProcessor.postProcessBeforeInitialization(ConfigurationPropertiesBindingPostProcessor.java:250)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsBeforeInitialization(AbstractAutowireCapableBeanFactory.java:408)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1566)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:539)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:476)
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:303)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:299)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:194)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:755)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:757)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:480)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:686)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:320)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:957)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:946)
	at com.capitalone.dashboard.Application.main(Application.java:16)
	... 6 more
Caused by: java.lang.IllegalStateException: Required Encryption configuration property missing: jasypt.encryptor.password
	at com.ulisesbocchio.jasyptspringboot.configuration.StringEncryptorConfiguration.getRequiredProperty(StringEncryptorConfiguration.java:80)
	at com.ulisesbocchio.jasyptspringboot.configuration.StringEncryptorConfiguration.lambda$static$0(StringEncryptorConfiguration.java:41)
	at com.ulisesbocchio.jasyptspringboot.encryptor.LazyStringEncryptor$SingletonFunction.lambda$new$1(LazyStringEncryptor.java:50)
	at com.ulisesbocchio.jasyptspringboot.encryptor.LazyStringEncryptor$SingletonFunction.apply(LazyStringEncryptor.java:63)
	at com.ulisesbocchio.jasyptspringboot.encryptor.LazyStringEncryptor.decrypt(LazyStringEncryptor.java:32)
	at org.jasypt.properties.PropertyValueEncryptionUtils.decrypt(PropertyValueEncryptionUtils.java:72)
	at com.ulisesbocchio.jasyptspringboot.EncryptablePropertySource.getProperty(EncryptablePropertySource.java:19)
	at com.ulisesbocchio.jasyptspringboot.wrapper.EncryptableMapPropertySourceWrapper.getProperty(EncryptableMapPropertySourceWrapper.java:28)
	at org.springframework.boot.bind.PropertySourcesPropertyValues.getEnumerableProperty(PropertySourcesPropertyValues.java:140)
	at org.springframework.boot.bind.PropertySourcesPropertyValues.processEnumerablePropertySource(PropertySourcesPropertyValues.java:124)
	at org.springframework.boot.bind.PropertySourcesPropertyValues.processPropertySource(PropertySourcesPropertyValues.java:106)
	at org.springframework.boot.bind.PropertySourcesPropertyValues.<init>(PropertySourcesPropertyValues.java:94)
	at org.springframework.boot.bind.PropertiesConfigurationFactory.getPropertyValues(PropertiesConfigurationFactory.java:281)
	at org.springframework.boot.bind.PropertiesConfigurationFactory.doBindPropertiesToTarget(PropertiesConfigurationFactory.java:252)
	at org.springframework.boot.bind.PropertiesConfigurationFactory.bindPropertiesToTarget(PropertiesConfigurationFactory.java:227)
	at org.springframework.boot.context.properties.ConfigurationPropertiesBindingPostProcessor.postProcessBeforeInitialization(ConfigurationPropertiesBindingPostProcessor.java:296)
	... 23 more`

**Comment by rob-miller-777** [comment]
@Mpluya , can you help us out here?

**Comment by vanjos** [comment]
Hi guys,

Anyone think they can help with this?

**Comment by Mpluya** [comment]
@vanjos it looks like you're using an encrypted property in your gitlab collector. Given this, you will need to pass the 'secret' in order for jasypt to decrypt the property value. In your docker run command, pass in an the 'JASYPT_ENCRYPTOR_PASSWORD' environment variable (e.g. `-e "JASYPT_ENCRYPTOR_PASSWORD=hygieiasecret"`). You may need to change hygieiasecret to whatever it is  that you used to encrypt your property value. 

If you continue to run into problems, please open up a different issue.

**Comment by vanjos** [comment]
That change helped, it now started.

I have moved the issue over to a new issue here #1318 as I am having issues with it connecting to Gitlab at all (seems to think that my repo does not exist).
